src/client/ua_client.c

```diff
@@ -20,6 +21,15 @@
 static void UA_Client_init(UA_Client* client, UA_ClientConfig config) {
     memset(client, 0, sizeof(UA_Client));
     client->channel.connection = &client->connection;
+    client->channel.endpoint = (UA_Endpoint*)UA_malloc(sizeof(UA_Endpoint));
```

Comment:
The test for NULL doesn't make any sense here in my opinion, since we have no way of notifying the caller, that something went wrong. If we make this return an error code, then sure. But i think we should maybe leave further client changes for a new PR, or not?